### DISCUSSION TITLE
How to do widget cleanup code

### DISCUSSION THREAD

**Discussion by andrewgoz** [TiddlyWiki/TiddlyWiki5#7695] [Q&A]
So, my widget needs to do some tidying before being thrown away. This is what I have now:
```
myWidget.prototype.origRemoveLocalDomNodes = myWidget.prototype.removeLocalDomNodes;
myWidget.prototype.removeLocalDomNodes = function() {
	// ... special handling here ...
	this.origRemoveLocalDomNodes();
};
```
It appears to work as intended. Is this the correct/best idiom for writing this?

**Comment by andrewgoz** [comment]
At the risk of answering myself, I've found the following has a better "feel":
```
myWidget.prototype.removeLocalDomNodes = function() {
	// ... special handling here ...
	Object.getPrototypeOf(Object.getPrototypeOf(this)).removeLocalDomNodes.call(this);
};
```
I'm still not sure if it is the "right" way or not.

**Reply by linonetwo** [reply]
Just do it in the `destroy()` https://tiddlywiki.com/dev/#Widget%20%60destroy%60%20method%20examples

It is temporarily removed in 5.3.1, but will be add back later.

**Reply by andrewgoz** [reply]
I'm currently using `removeLocalDomNodes()` which is working well as far as I can tell. Is there a benefit to switching to `destroy()`?

**Reply by linonetwo** [reply]
I think they are both temporarily removed in 5.3.1, Does it still exist?

**Reply by andrewgoz** [reply]
You're right, that's gone in 5.3.1, but it is present in the 5.3.0 I'm using. I'll stick with 5.3.0 until the question of `destroy()`/`removeLocalDomNodes()` is resolved.

**Comment by linonetwo** [comment]
See https://github.com/Jermolene/TiddlyWiki5/discussions/5945#discussioncomment-8353993 , use `@wessberg/connection-observer`.

Or use custom element, if you can figure out how to use it.
